crates/accelerate/src/twirling.rs

```diff
@@ -123,12 +128,59 @@ const CZ_MASK: u8 = 4;
 const ECR_MASK: u8 = 2;
 const ISWAP_MASK: u8 = 1;
 
+#[inline(always)]
+fn diff_frob_norm(array: ArrayView2<Complex64>, gate_matrix: ArrayView2<Complex64>) -> f64 {
+    let mut res: f64 = 0.;
+    for i in 0..4 {
+        for j in 0..4 {
+            let gate = gate_matrix[[i, j]];
+            let twirled = array[[i, j]];
+            let diff = twirled - gate;
+            res += (diff.conj() * diff).re;
+        }
+    }
+    res.sqrt()
+}
+
+fn generate_twirling_set(gate_matrix: ArrayView2<Complex64>) -> Vec<([StandardGate; 4], f64)> {
+    let mut out_vec = Vec::with_capacity(16);
+    let i_matrix = aview2(&ONE_QUBIT_IDENTITY);
+    let x_matrix = aview2(&qiskit_circuit::gate_matrix::X_GATE);
+    let y_matrix = aview2(&qiskit_circuit::gate_matrix::Y_GATE);
+    let z_matrix = aview2(&qiskit_circuit::gate_matrix::Z_GATE);
+    let iter_set = [
+        (IGate, i_matrix),
+        (XGate, x_matrix),
+        (YGate, y_matrix),
+        (ZGate, z_matrix),
+    ];
+    for (i, i_mat) in &iter_set {
+        for (j, j_mat) in &iter_set {
+            let before_matrix = kron(j_mat, i_mat);
+            let half_twirled_matrix = gate_matrix.dot(&before_matrix);
+            for (k, k_mat) in &iter_set {
+                for (l, l_mat) in &iter_set {
+                    let after_matrix = kron(l_mat, k_mat);
```

Comment:
If we store the 16 possible krons outside all the loops, we can avoid recalculating the same krons 16 times over in here.  It's a factor of 17 saving in the number of krons (though the 2q matmuls will still be far more expensive - kron is really SIMD and cache-line friendly, and it's a lot fewer scalar ops than a matmul).

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13331/commits/aaa62951e9ddf0fca358a269a463a3e2fd42daa7